00_Base/src/repository/TokensRepository.ts

```diff
@@ -285,15 +312,79 @@ export class TokensRepository extends SequelizeRepository<OcpiToken> {
       throw new UnknownTokenException('Token not found in the database');
     }
 
-    const updatedToken = updatedTokens?.[0];
+    return updatedTokens[0];
+  }
 
-    if (updatedCount > 1) {
+  private async updateAuthorization(
+    existingAuth: Authorization,
+    partialToken: Partial<TokenDTO>,
+    transaction: SequelizeTransaction,
+  ): Promise<Authorization> {
+    const partialIdTokenInfo = this.mapTokenDTOToPartialAuthorization(
+      existingAuth,
+      partialToken,
+    );
+
+    if (!partialIdTokenInfo) {
+      return existingAuth;
+    }
+
+    const [updateCount, updatedIdTokenInfo] = await IdTokenInfo.update(
+      partialIdTokenInfo,
+      {
+        where: { id: existingAuth.idTokenInfoId },
+        returning: true,
+        transaction,
+      },
+    );
+
+    if (updateCount === 0) {
       this.logger.warn(
-        `More than one token updated for primary ID ${existingOcpiToken.id}`,
+        `No updatable idTokenInfo found for auth ${existingAuth.id}`,
       );
+      return existingAuth;
+    }
+
+    if (partialIdTokenInfo.groupIdToken) {
+      await IdToken.update(partialIdTokenInfo.groupIdToken, {
+        where: { id: updatedIdTokenInfo[0].groupIdTokenId },
+      });
+    }
+
+    return await existingAuth.reload({
+      include: [{ model: IdTokenInfo }],
+      transaction,
+    });
+  }
+
+  private mapTokenDTOToPartialAuthorization(
+    existingAuth: Authorization,
+    tokenDTO: Partial<TokenDTO>,
+  ): Partial<IdTokenInfoType> {
+    const idTokenInfo: IdTokenInfoType = {
+      status: existingAuth.idTokenInfo?.status!,
```

Comment:
linter error her, seems that it doesn't like the not null assertion here 🤔 i think it's solvable by making the type of id token into a `Partial<IdTokenInfoType>` instead...?